### PR TITLE
How to import and create modules.

### LINKED ISSUE #2617: MERN_NodeJs 4.1 How to import and create modules?
# Instructions

## Project directory
Branch MERN>Web_Development/MERN/Topic/Backend/Node/

FileName: 4.1_Import and create modules.md

## How to import and create modules?
Explain how to how to import and create modules.

# Task

- [ ]  Audio
- [ ]  Documentation
- [ ]  Video

## Send a PR in the following branch only and make documentation in form of markdown file with (.md) extension and in the respective folder

### PR DESCRIPTION
**Pull request by WajihaFatim**
<hr>

## Description 📜

Fixes #2617

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [x] MERN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by WajihaFatim** [comment]
@EkjotKaur please check

File: MERN/Topic/Backend/Node/4.1_Import and create modules.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+In Node. js, Modules are the blocks of encapsulated code that communicates with an external application on the basis of their related functionality
+ Modules can be a single file or a collection of multiples files/folders
+## Why do we use it?
+The major advantage of Modularity is reusability. We don’t need to redevelop the same existing functionality. We can import a Module and reuse that module functionality very easily.
+
+## IMPORTING NODEjs MODULES
+First we install the module using command in the terminal in the your project directory .  
+
+          npm install module_name
```
Comment (Naman-1234) [inline MERN/Topic/Backend/Node/4.1_Import and create modules.md:11]: Please also mention about other packages as well, Example `yarn add module_name`.

File: MERN/Topic/Backend/Node/4.1_Import and create modules.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+### SYNTAX :
+     
+
+          var some-name= require('module-name')     
+      
+         
+
+### EXAMPLE
+•	Import Nodejs built-in “express” module.
```
Comment (Naman-1234) [inline MERN/Topic/Backend/Node/4.1_Import and create modules.md:35]: Correction: express and mongoose are not built-in modules, They both have to be installed before using them.

File: MERN/Topic/Backend/Node/4.1_Import and create modules.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+![Screenshot (283)](https://user-images.githubusercontent.com/68144616/134819000-b9ef8f9a-93fd-4548-8ba5-7d9ff9cdb167.png)
+
+
+Now Lets see,
+
+
+## HOW TO CREATE A NODEjs MODULE
+
+Step 1 : Create a package.json file using the command below in your directory
```
Comment (Naman-1234) [inline MERN/Topic/Backend/Node/4.1_Import and create modules.md:58]: Also, mention command `npm init -y` for setting all values to default.

**Review by Naman-1234 — CHANGES_REQUESTED** [review]

File: MERN/Topic/Backend/Node/4.1_Import and create modules.md (hunk 3 above)
Comment (WajihaFatim) [reply MERN/Topic/Backend/Node/4.1_Import and create modules.md:58]: okay

**Comment by EkjotKaur** [comment]
@WajihaFatim Please change the name from 4.1 to 5.2.
It was my mistake . I had written incorrectly in the Issue. Please make the changes.

**Comment by WajihaFatim** [comment]
Do I have to make the new pr for this

**Pull request merged**